msbuild/Xamarin.MacDev.Tasks.Core/Tasks/AlToolTaskBase.cs

```diff
@@ -82,5 +96,28 @@ string GetFileTypeValue ()
 				default: throw new NotSupportedException ($"Provided file type '{FileType}' is not supported by altool");
 			}
 		}
+
+		void LogErrorsFromOutput (string output)
+		{
+			try {
+				if (string.IsNullOrEmpty(output))
+					return;
+
+				var plist = PObject.FromString (output) as PDictionary;
+				var errors = PObject.Create (PObjectType.Array) as PArray;
+				var message = PObject.Create (PObjectType.String) as PString;
+
+				if ((plist?.TryGetValue ("product-errors", out errors) == true)) {
+					foreach (var error in errors) {
+						var dict = error as PDictionary;
+						if (dict?.TryGetValue ("message", out message) == true) {
+							Log.LogError (ToolName, null, null, null, 0, 0, 0, 0, "{0}", message.Value);
```

Comment:
not sure if the API (or some overloads) allows it but `"{0}"` seems to require a call to `String.Format`, can't you just provide `message.Value` ?

Reply:
@spouliot What if `message.Value` contains something that can be a format string? Say `"{0}"` for instance, then things might fail (depending on LogError's implementation).

Reply:
yeah, bring us back to _not sure if the API_ :)

Reply:
yes, I don't really like it (using `"{0}"`) but it seems to be _safer_, and that's how most of the XI tasks log errors that come from tool outputs or text we don't control.